File: src/main/java/it/contrader/dto/UserDTO.java
package it.contrader.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import it.contrader.model.User.Usertype;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class UserDTO {

	private long id;
	
	private String username;
	
	private String password;

	private Usertype usertype;
	
	private DriverDTO driver;  
	
	private PassengerDTO passenger;
	
	
	private boolean active = true;
	

}

File: src/dto/userdto.ts
import { Usertype } from './usertype';
import { DriverDTO } from './driverdto';
import { PassengerDTO } from './passengerdto';
/**
 * Classe DTO di User. DEVE essere uguale (stesso nome classe, stessi attributi e stessi nomi) a
 * quello nel backend.
 *
 * @see Usertype
 *
 * @author Vittorio Valent
 */
export class UserDTO {

   id: number;

   username: string;

   password: string;

   usertype: Usertype;

   driver: DriverDTO;

   passenger: PassengerDTO;

   active = true;

   setid(id:number){
     this.id = id;
   }


}

